client/src/components/CalendarPage.js

```diff
@@ -20,64 +19,29 @@ class CalendarPage extends Component {
   state = {
     schedulingModal: false,
     checkboxModal: false,
-    performances: [],
+    focusedPerformances: [],
+    weight: "",
+    sets: "",
+    reps: "",
     usageMode: "NEW_WORKOUT", // or COPY_WORKOUTS
     copyFromStartDate: "",
     copyFromEndDate: "",
-    copyToStartDate: ""
+    copyToStartDate: "",
   };
 
-  /* This is for putting the performances array from the calendar reducer onto local state.
-  this.state.performances is being used to toggle completed field of each Performance Doc*/
-  static getDerivedStateFromProps(props, state) {
-    if (props.performances !== state.performances) {
-      return {
-        performances: props.performances
-      };
-    }
-    // Return null if the state hasn't changed
-    return null;
-  }
-
   componentDidMount() {
     this.props.fetchRoutines();
     this.props.fetchAllWorkouts();
-    this.props.fetchAllPerformanceDocs();
   }
 
+  //-------------------------------------------- callback functions for scheduling modal
```

Comment:
Good comment. We should spend some time as a group (tomorrow morning)? reading through the codebase together and writing commentary.